plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -353,16 +355,16 @@ else if (precision > Decimals.MAX_PRECISION || actualPrecision <= 0) {
 
             // This mapping covers both DATE and TIMESTAMP, as Oracle's DATE has second precision.
             case OracleTypes.TIMESTAMP:
-                if ("DATE".equalsIgnoreCase(typeHandle.getJdbcTypeName().orElseThrow(() -> new IllegalStateException("type name not present")))) {
+                if (typeHandle.getDecimalDigits().isEmpty()) {
```

Comment:
While i suggested checking decimal digits, i hoped for something explit like

```java
if (typeHandle.getDecimalDigits() == 0) {
  // it's a date
  ....
}

// it's a timestamp
...
```

checking `typeHandle.getDecimalDigits().isEmpty()` looks like checking something that could change in the future, without notice.

Let's return to investigating the `typeHandle.getJdbcTypeName()`